### ISSUE TITLE
BraveBeta.exe process not always being terminated when closing Brave via title bar

### ISSUE THREAD

**Issue by kjozwiak** [brave/browser-laptop#13233]
### Description

@btlechowski (great find BTW 👍) found that when closing brave via the 'X' button under the title bar, the `BraveBeta.exe` process doesn't terminate which can lead to situations where clicking on the `Brave` shortcut on the desktop doesn't launch the application due to the process still running.

### Test Plan / Steps to Reproduce

  1. launch `0.21.11` and open your favourite process explorer 
  2. once launched, close brave using the `X` located on the title bar
  3. relaunch Brave

You'll notice that several `BraveBeta.exe` processes haven't been closed and are still running.

**Actual result:**

![winprocesses](https://user-images.githubusercontent.com/2602313/36503959-ed03d30a-171c-11e8-90f7-ab6d98a8b926.gif)

<img width="767" alt="screen shot 2018-02-21 at 3 01 35 pm" src="https://user-images.githubusercontent.com/2602313/36503966-f28b2198-171c-11e8-979f-4cf918b454b0.png">

**Expected result:**

All `BraveBeta.exe` process should be terminated correctly when Brave is closed.

**Reproduces how often:**

100% reproducible when following the STR mentioned above.

### Brave Version

**about:brave info:**

```
Brave: 0.21.11 
V8: 6.4.388.41 
rev: e146cd88d6cc5c91f8e772a18cd6c604fe903f58 
Muon: 5.0.5 
OS Release: 10.0.16299 
Update Channel: Beta 
OS Architecture: x64 
OS Platform: Microsoft Windows 
Node.js: 7.9.0 
Brave Sync: v1.4.2 
libchromiumcontent: 64.0.3282.140
```

**Reproducible on current live release:**

Couldn't reproduce with the current release.

**Comment by btlechowski** [comment]
I am able to reproduce with 100% rate using these steps:
 Steps:
 1. Clean install
 2. Open Brave
 3. Close Brave using titlebar's Close button (x)
 4. Check running processes for BraveBeta

Tested on Win7 x64
```
Brave: 0.21.11 
V8: 6.4.388.41 
rev: e146cd88d6cc5c91f8e772a18cd6c604fe903f58 
Muon: 5.0.5 
OS Release: 6.1.7601 
Update Channel: Beta 
OS Architecture: x64 
OS Platform: Microsoft Windows 
Node.js: 7.9.0 
Brave Sync: v1.4.2 
libchromiumcontent: 64.0.3282.140
```

**Comment by kjozwiak** [comment]
Reproduced this on Ubuntu 17.10 x64 using `0.21.11` as well. Example:

<img width="945" alt="screen shot 2018-02-21 at 3 56 23 pm" src="https://user-images.githubusercontent.com/2602313/36504846-d24468d8-171f-11e8-8377-a5b61151d9da.png">

**Comment by btlechowski** [comment]
On Ubuntu, when BraveBeta is launched through Brave Beta shortcut on TaskBar, the browser is actually recognized as Brave and not Brave Beta(Check Brave shortcut). 

![image](https://user-images.githubusercontent.com/34715963/36505288-4e2fe020-1753-11e8-8aee-d46febba265b.png)

**Comment by darkdh** [comment]
reproducible using muon 5.0.0 and 4.8.2 so might be something wrong on browser-laptop

**Comment by petemill** [comment]
Probably related to bufferWindow. We should exclude the buffer window from checking if all windows are closed.

**Comment by petemill** [comment]
Thanks for finding @btlechowski and @kjozwiak! A potential fix is pending in #13242.

**Comment by srirambv** [comment]
Verified on Windows x64 
* 0.22.6  e6ff4ea
* libchromiumcontent: 65.0.3325.162
* muon: 5.1.0
